src/test/java/mil/dds/anet/test/resources/PersonResourceTest.java

```diff
@@ -562,6 +567,66 @@ public void testInsertCustomSensitiveInformation()
         true);
   }
 
+  @Test
+  public void testUpdatePersonHistory() throws Exception {
+    final OrganizationSearchQueryInput query = OrganizationSearchQueryInput.builder()
+        .withText("EF 6").withType(OrganizationType.ADVISOR_ORG).build();
+    final AnetBeanList_Organization orgs =
+        jackQueryExecutor.organizationList(getListFields("{ uuid shortName }"), query);
+    assertThat(orgs.getList().size()).isGreaterThan(0);
+    final Organization org = orgs.getList().stream()
+        .filter(o -> o.getShortName().equalsIgnoreCase("EF 6")).findFirst().get();
+    assertThat(org.getUuid()).isNotNull();
+
+    final PersonInput persInput = PersonInput.builder().withRole(Role.ADVISOR)
+        .withName("Test person for edit history").build();
+    Person person = adminMutationExecutor.createPerson(FIELDS, persInput);
+    assertThat(person).isNotNull();
+    assertThat(person.getUuid()).isNotNull();
+    // Create a Position
+    final PositionInput testInput1 = PositionInput.builder().withType(PositionType.ADVISOR)
+        .withName("Test Position for person history edit  1")
+        .withOrganization(getOrganizationInput(org))
+        .withLocation(getLocationInput(getGeneralHospital())).withStatus(Status.ACTIVE).build();
+
+    final Position createdPos1 = adminMutationExecutor.createPosition(POSITION_FIELDS, testInput1);
+    assertThat(createdPos1).isNotNull();
+    assertThat(createdPos1.getUuid()).isNotNull();
+    assertThat(createdPos1.getName()).isEqualTo(testInput1.getName());
+    PositionInput posInput1 = PositionInput.builder().withUuid(createdPos1.getUuid()).build();
+    final PositionInput testInput2 = PositionInput.builder().withType(PositionType.ADVISOR)
+        .withName("Test Position for person history edit 2")
+        .withOrganization(getOrganizationInput(org))
+        .withLocation(getLocationInput(getGeneralHospital())).withStatus(Status.ACTIVE).build();
+
+    final Position createdPos2 = adminMutationExecutor.createPosition(POSITION_FIELDS, testInput2);
+    assertThat(createdPos2).isNotNull();
+    assertThat(createdPos2.getUuid()).isNotNull();
+    assertThat(createdPos2.getName()).isEqualTo(testInput2.getName());
+    PositionInput posInput2 = PositionInput.builder().withUuid(createdPos2.getUuid()).build();
+    PersonPositionHistoryInput hist1 = PersonPositionHistoryInput.builder()
+        .withCreatedAt(Instant.now().minus(100, ChronoUnit.DAYS))
+        .withStartTime(Instant.now().minus(100, ChronoUnit.DAYS))
+        .withEndTime(Instant.now().minus(50, ChronoUnit.DAYS)).withPosition(posInput1).build();
+    PersonPositionHistoryInput hist2 = PersonPositionHistoryInput.builder()
+        .withCreatedAt(Instant.now().minus(100, ChronoUnit.DAYS))
+        .withStartTime(Instant.now().minus(100, ChronoUnit.DAYS))
+        .withEndTime(Instant.now().minus(50, ChronoUnit.DAYS)).withPosition(posInput2).build();
```

Comment:
These two history entries overlap, so the test should fail. Since the test passes, the check logic in `ResourceUtils.java` appears to be incorrect.